test/date-time/date-time-utility.test.ts

```diff
@@ -0,0 +1,397 @@
+                () =>
+                {
+                    Assert.strictEqual(daysOfMonth.length, 29);
+                }
+            );
+
+            test(`Given a DateTime object with value in leap year February "2024-02-01 10:00"
+            when daysOfMonth is calculated for that DateTime
+            then first element of the array should represent first day of month`,
+                () =>
+                {
+                    Assert.strictEqual(daysOfMonth.takeFirst().value, "2024-02-01 00:00");
+                    Assert.strictEqual(daysOfMonth.takeFirst().dateValue, "2024-02-01");
+                }
+            );
+
+            test(`Given a DateTime object with value in leap year February "2024-02-01 10:00"
+            when daysOfMonth is calculated for that DateTime
+            then last element of the array should represent last day of month`,
+                () =>
+                {
+                    Assert.strictEqual(daysOfMonth.takeLast().value, "2024-02-29 23:59");
+                    Assert.strictEqual(daysOfMonth.takeLast().dateValue, "2024-02-29");
+                }
+            );
+        });
+
+        suite("for 2023 feb (non leap year)", () =>
+        {
+            const daysOfMonth = new DateTime({ value: "2023-02-01 10:00", zone: "utc" }).getDaysOfMonth();
+
+            test(`Given a DateTime object with value in non leap year February "2023-02-01 10:00"
+            when daysOfMonth is calculated for that DateTime
+            then it should return an array of 28 days`,
+                () =>
+                {
+                    Assert.strictEqual(daysOfMonth.length, 28);
+                }
+            );
+
+            test(`Given a DateTime object with value in non leap year February "2024-02-01 10:00"
+            when daysOfMonth is calculated for that DateTime
+            then first element of the array should represent first day of month`,
+                () =>
+                {
+                    Assert.strictEqual(daysOfMonth.takeFirst().value, "2023-02-01 00:00");
+                    Assert.strictEqual(daysOfMonth.takeFirst().dateValue, "2023-02-01");
+                }
+            );
+
+            test(`Given a DateTime object with value in non leap year February "2024-02-01 10:00"
+            when daysOfMonth is calculated for that DateTime
+            then last element of the array should represent last day of month`,
+                () =>
+                {
+                    Assert.strictEqual(daysOfMonth.takeLast().value, "2023-02-28 23:59");
+                    Assert.strictEqual(daysOfMonth.takeLast().dateValue, "2023-02-28");
+                }
+            );
+        });
+    });
+
+    suite("Convert to zone", () =>
+    {
+        const value = "2024-01-01 10:00";
+
+        const dateTime = new DateTime({ value, zone: "utc" });
+
+        test(`Given a DateTime object with value (${value}) and zone (utc)
```

Comment:
Have zone coversion test for the Daylist saving times is on vs not on for a particular zone

Reply:
?